qubesmanager/qube_manager.py

```diff
@@ -61,140 +70,363 @@ def mousePressEvent(self, e):  # pylint: disable=invalid-name
+        self.name = self.vm.name
+        self.label = self.vm.label
+        self.klass = self.vm.klass
+        self.state = {'power': "", 'outdated': ""}
+        self.updateable = getattr(vm, 'updateable', False)
+        self.update(True)
 
     def update(self, update_size_on_disk=False, event=None):
         """
-        Update info in a single VM row
-        :param update_size_on_disk: should disk utilization be updated? the
-        widget will extract the data from VM object
+        Update VmInfo
+        :param update_size_on_disk: should disk utilization be updated?
         :param event: name of the event that caused the update, to avoid
         updating unnecessary properties; if event is none, update everything
         :return: None
         """
         try:
-            self.info_widget.update_vm_state()
+            self.state['power'] = self.vm.get_power_state()
+
+            if self.vm.is_running():
+                if hasattr(self.vm, 'template') and \
+                        self.vm.template.is_running():
+                    self.state['outdated'] = "to-be-outdated"
+                else:
+                    for vol in self.vm.volumes.values():
+                        if vol.is_outdated():
+                            self.state['outdated'] = "outdated"
+                            break
+            else:
+                self.state['outdated'] = ""
+
+            if self.vm.klass in {'TemplateVM', 'StandaloneVM'} and \
+                    self.vm.features.get('updates-available', False):
+                self.state['outdated'] = 'update'
+
             if not event or event.endswith(':label'):
-                self.label_widget.update()
+                self.label = self.vm.label
             if not event or event.endswith(':template'):
-                self.template_widget.update()
+                try:
+                    self.template = self.vm.template.name
+                except exc.QubesNoSuchPropertyError:
+                    self.template = self.vm.klass
             if not event or event.endswith(':netvm'):
-                self.netvm_widget.update()
+                self.netvm = getattr(self.vm, 'netvm', None)
+                if self.netvm:
+                    self.netvm = self.netvm.name
             if not event or event.endswith(':internal'):
                 # this is a feature, not a property; TODO: fix event handling
-                self.internal_widget.update()
+                if self.vm.features.get('internal', False):
+                    self.internal = "Yes"
+                else:
+                    self.internal = ""
             if not event or event.endswith(':ip'):
-                self.ip_widget.update()
+                self.ip = getattr(self.vm, 'ip', None)
             if not event or event.endswith(':include_in_backups'):
-                self.include_in_backups_widget.update()
+                self.inc_backup = getattr(self.vm, 'include_in_backups', None)
```

Comment:
Bool properties better present as `Yes` / empty, as you've preserved it in `internal` property.